### PR TITLE
BUG: Panel setitem with a multiindex #10360 (partial)

### PR DESCRIPTION
**Pull request by jreback**
partial on #10360

### CODE REVIEW

**Comment by jreback** [comment]
cc @seth-p

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -213,6 +214,16 @@ def _setitem_with_indexer(self, indexer, value):
...
+        if isinstance(indexer, tuple) and len(indexer) == len(self.obj.axes):
+
+            for i, ax in zip(indexer, self.obj.axes):
+
+                # if we have any multi-indexes that have non-trivial slices (not null slices)
+                # then we must take the split path, xref GH 10360
+                if isinstance(ax, MultiIndex) and not (is_integer(i) or is_null_slice(i)):
+                    take_split_path = True
+                    break
+
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: @jreback looks like this is intended for Panel, but is still reached.  Any chance it is unnecessary?  Or more specifically, I'd like to rule out the possibility of `take_split_path = True` being set on 224 for `Series`.  Is that unreachable for some reason?

Comment (jreback) [reply pandas/core/indexing.py]: umm maybe, the whole take_split_path is based on the notion that we have a block  of a single type and we are indexing into it for smaller number of columns than the whole block so we *might* need to split it before we index. fairly dense & complicated, but necessary in some way.

to answer your question though, I agree you might be able to remove this entire clause.

Comment (jorisvandenbossche) [reply pandas/core/indexing.py]: Apparently there was still a use case: https://github.com/pandas-dev/pandas/pull/27855

**Pull request merged**